fdbrpc/sim2.actor.cpp

```diff
@@ -1688,15 +1688,20 @@ class Sim2 final : public ISimulator, public INetworkConnections {
 	}
 	void killProcess(ProcessInfo* machine, KillType kt) override {
 		TraceEvent("AttemptingKillProcess").detail("ProcessInfo", machine->toString());
-		if (kt < RebootAndDelete) {
+		// Refuse to kill a protected process.
```

Comment:
I could not find any info on what a protected process is. What happens if we do kill a protected process? What is the testing opportunity lost, if any, from making it so certain processes cannot be killed (I imagine that in a real cluster, the machine on which such a process is running could lose power).

Reply:
I guess "what happens if we do kill" is a function of what process it is. If the essential property of a protected address is that it is one where failure cannot be induced, then maybe it should just be renamed to immortalAddress. Either way, some description of the concept would be useful - maybe in simulator.h, at the declaration.

Reply:
My understanding of protected process is that it should be used for testers. Killing a tester is almost never useful and therefore we want to stop machine attrition doing that.

So I would like to understand why this specific process is protected

Reply:
I've gathered that protected processes are those which we're telling the simulator or workload not to randomly kill, because the cluster may be unrecoverable if we do. Specifically, on simulated cluster startup, we mark a random majority of the coordinators as protected, to avoid accidentally killing a quorum of them.

In any case, I added the checks to these two functions because killProcess_internal was asserting that we were not attempting to kill a protected process.

I don't think we're missing any testing opportunities here. It's not always possible to kill the machines we've randomly selected. The simulation is conservative in the chaos it causes so that the errors we get out of simulation testing are real.

Reply:
I've seen that workloads that exercise attrition are able to specify a min number of machines that must remain running. That doesn't say what kind - so the requirement that the simulated cluster never lose quorum of coordinators is implicit, which makes sense. I would request that we document (as a comment in simulator.h or wherever else you deem fit) that the we're trying to keep simulation reasonable and realistic when inducing failure.

Reply:
Added some comments about protectedAddresses.